### PR TITLE
Support clone()/fork()/exec() in runtime

### PR DESCRIPTION
**Pull request by fw-immunant**
We need `clone()` to sandbox multithreaded programs; `fork()`/`exec()` support is needed for Criterion tests as well as any sandboxed program that wants to spawn a child.

This raises some questions: seccomp filters are *necessarily* inherited, and if our ptrace() tracer doesn't attend to forked child processes, they will see any `SECCOMP_RET_TRACE` syscalls as forbidden and will generally not work. As such, our tracer must be responsible for *all* child processes, even if they aren't actually sandboxed with IA2.

However, we also have the situation (with Criterion) where the child process *is* sandboxed with IA2. For now, we simply try to treat all child processes like IA2-sandboxed ones.

~This is a draft because it's still pretty messy as it's where things stood when I gave up on making Criterion work (for some reason one of its child processes gets ENOENT when trying to open a shm object that was created by another process... the shm object does indeed not exist, but I don't know why this is the case under our sandbox and not otherwise).~

I've gotten this to work with most of our tests, the exceptions being `threads`, `protected_threads`, and `permissive_mode`. The largest problem was that we were not handing ptrace "group-stop" correctly, and we couldn't until switching from `PTRACE_TRACEME` to `PTRACE_SEIZE`.

### CODE REVIEW

File: cmake/define-test.cmake [hunk 1]
```diff
@@ -142,7 +142,7 @@ endfunction()
 #                  cmake test infrastructure.
 function(define_test)
     # Parse options
-    set(options NEEDS_LD_WRAP NOT_IN_CHECK_IA2 NO_LIBS CRITERION_TEST)
+    set(options NEEDS_LD_WRAP NOT_IN_CHECK_IA2 NO_LIBS CRITERION_TEST WITHOUT_RUNTIME)
```
Comment (rinon) [inline cmake/define-test.cmake:145]: Please add to doc string

Comment (rinon) [reply cmake/define-test.cmake:145]: Also, what do you think about `WITHOUT_SANDBOX`?

File: runtime/seccomp_filter.c [hunk 2]
```diff
@@ -47,22 +48,71 @@ struct sock_filter ia2_filter[] = {
...
+    BPF_SYSCALL_POLICY(geteuid, ALLOW),
+    BPF_SYSCALL_POLICY(getgid, ALLOW),
+    BPF_SYSCALL_POLICY(getuid, ALLOW),
+    BPF_SYSCALL_POLICY(getppid, ALLOW),
+    BPF_SYSCALL_POLICY(setpgid, ALLOW), // used by criterion
+    BPF_SYSCALL_POLICY(gettid, ALLOW),
     BPF_SYSCALL_POLICY(getpid, ALLOW),
     BPF_SYSCALL_POLICY(getrandom, ALLOW),
+    BPF_SYSCALL_POLICY(prctl, ALLOW),
+    BPF_SYSCALL_POLICY(ioctl, ALLOW),
```
Comment (rinon) [inline runtime/seccomp_filter.c:67]: `ioctl` and friends deserves a comment at least, we can't just allow all ioctls through, can we? Or are we sandboxing what devices can be opened? Not saying we should do this now, just don't want to forget.

**Review by rinon — APPROVED** [review]

File: runtime/seccomp_filter.c (hunk 2 above)
Comment (fw-immunant) [reply runtime/seccomp_filter.c]: Good point; this is needed by Criterion but we can fairly easily limit it to only the one `ioctl` that Criterion actually performs.

Comment (fw-immunant) [reply runtime/seccomp_filter.c]: Done in b449ff0.

File: cmake/define-test.cmake (hunk 1 above)
Comment (fw-immunant) [reply cmake/define-test.cmake]: Done in d542238.

**Pull request merged**